evaluate/task/task.go

```diff
@@ -47,3 +50,35 @@ func TaskForIdentifier(taskIdentifier evaltask.Identifier) (task evaltask.Task,
 		return nil, pkgerrors.Wrap(evaltask.ErrTaskUnknown, string(taskIdentifier))
 	}
 }
+
+// taskLogging holds common logging functionality.
+type taskLogging struct {
+	log      *log.Logger
+	logClose func()
+	ctx      evaltask.Context
+	task     evaltask.Task
+}
```

Comment:
```go
type taskLogger struct {
	*log.Logger

	logClose func()
	ctx      evaltask.Context
	task     evaltask.Task
}
```

Then you still have "initialize" and "finalize" but use the struct as regular logger.
Then maybe rename "initializeLogging" to `NewTaskLogger`?!

Reply:
Will not export though to prevent misuse.